lib/jekyll/commands/console.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require "irb"
+
+module Jekyll
+  module Commands
+    class Console < Command
+      class << self
+        def init_with_program(prog)
+          prog.command(:console) do |c|
+            c.syntax "console"
+            c.description "Invoke an IRB console with the site loaded"
+            c.alias :c
+
+            c.option "config", "--config CONFIG_FILE[,CONFIG_FILE2,...]", Array,
+                     "Custom configuration file"
+
+            c.action do |_, options|
+              Jekyll::Commands::Console.process(options)
+            end
+          end
+        end
+
+        # TODO: is there a way to add a unit test for this command?
+        # rubocop:disable Style/GlobalVars, Metrics/AbcSize, Metrics/MethodLength
+        def process(options)
+          Jekyll.logger.info "Starting:", "Jekyll v#{Jekyll::VERSION.magenta}" \
+                                      " console…"
+          Jekyll.logger.info "Environment:", Jekyll.env.cyan
+          site = Jekyll::Site.new(configuration_from_options(options))
+          site.reset
+          site.read
+          site.generate
```

Comment:
I feel the user would like to know that some of the build phases have already been run..
For example,
```
Reading files... done!
Running generators... done!
```
In large sites, there will be a significant time delay between `site.reset` and `site.render`. So, we don't want the use reporting back to us that *the console takes a long time to boot..*

That said, IMHO, I'd like an option to start the console with just `Jekyll::Site.new(config)` executed so that I can manually run `site.read` and `site.generate` etc at a later stage..

Reply:
Build phase notifications sounds good. For an option to skip those and only do the new/config, any thoughts? `--skip-build` or `--skip-load` or something like that?

Reply:
I like `--blank`.